sites/public/lib/helpers.tsx

```diff
@@ -64,6 +65,45 @@ export const getListings = (listings) => {
     rent: t("t.rent"),
     availability: t("t.availability"),
   }
+
+  const { profile, userProfileService } = useContext(AuthContext)
+
+  const addToFavorite = async (listingID: string) => {
```

Comment:
I think if we change the relation to be ManyToMany instead of json blob, then these functions can be simpler and just pass the ID to add to the backend and let the backend deal with adding the relation if it doesn't exist.